src/utils/Dimensions.ts

```diff
@@ -0,0 +1,33 @@
+module n3Charts.Utils {
+  'use strict';
+
+  export class Dimensions {
+    public width = 600;
+    public height = 200;
+    public innerWidth = 560;
+    public innerHeight = 160;
+    public margin = {
+      left: 30,
+      bottom: 20,
+      right: 20,
+      top: 20
+    };
+
+    getPixelCssProp(element: HTMLElement, propertyName: string) {
+      var style = window.getComputedStyle(element, null);
+      return +style.getPropertyValue(propertyName).replace(/px$/, '');
+    }
+
+    fromParentElement(parent: HTMLElement) {
+      // Oooooh I hate doing this.
```

Comment:
Hmmmm the parent, like the div generated by the directive ? And I'm not familiar with box-sizing, can you explain a bit more please ? :-)